salt/ca/init.sls

```diff
@@ -54,10 +54,12 @@ salt-minion:
 /etc/pki/ca.crt:
   x509.certificate_managed:
     - signing_private_key: /etc/pki/ca.key
-{% if salt['grains.get']('domain', None) is none %}
+{% if salt['grains.get']('domain', '')|length > 0 %}
+    - CN: {{ grains['domain'] }}
+{% elif salt['pillar.get']('dns:domain', '')|length > 0 %}
     - CN: {{ pillar['dns']['domain'] }}
 {% else %}
-    - CN: {{ grains['domain'] }}
+    - CN: kubernetes
```

Comment:
I like @inercia's proposal, but I'd like to make it more obvious that this is a default that we've created (maybe in the future we use another value, and when inspecting an "old" cluster we have this CN are we are clueless about it). So, what about something like: "caasp_default_cn" ?